### PR TITLE
[Closes #74] ktn_os:command/[1,2]

### LINKED ISSUE #74: Implement a ktn_os:command/[1, 2] function that uses ports
[eunit](https://github.com/erlang/otp/blob/maint/lib/eunit/src/eunit_lib.erl#L421) already has an implementation using this approach.

The function in `ktn_os` should provide an option to show in `stdout` the output from the execution of the command while it is being generated.

### PR DESCRIPTION
**Pull request by jfacorro**

### CODE REVIEW

File: src/ktn_os.erl [hunk 1]
```diff
@@ -0,0 +1,45 @@
+%% @doc Utility functions for
```
Comment (elbrujohalcon) [inline src/ktn_os.erl]: for … ? (trollface)

File: src/ktn_os.erl [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        undefined -> ok
+      end,
+      get_data(Port, Opts, [NewData | Data]);
+    {Port, eof} ->
+      port_close(Port),
+      receive
+        {Port, {exit_status, ExitStatus}} ->
+          {ExitStatus, lists:flatten(lists:reverse(Data))}
+      end
+  end.
```
Comment (elbrujohalcon) [inline src/ktn_os.erl]: Can we have an `after` clause here? The value should be in `Opts` with a default of, say… an hour?

Comment (elbrujohalcon) [inline src/ktn_os.erl]: Also, what happens if the shell process is killed? Do we get a different message? Maybe an error or an exit signal?

File: test/ktn_os_SUITE.erl [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+%% Common test
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+-spec all() -> [atom()].
+all() ->
+    Exports = ?MODULE:module_info(exports),
+    [F || {F, _} <- Exports, not lists:member(F, ?EXCLUDED_FUNS)].
+
+-spec init_per_suite(config()) -> config().
+init_per_suite(Config) ->
```
Comment (elbrujohalcon) [inline test/ktn_os_SUITE.erl]: Since they do nothing, I think you should just remove these functions.

File: test/ktn_os_SUITE.erl [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  {ok, Cwd} = file:get_cwd(),
+  Result = Cwd ++ "\n",
+  {0, Result} = ktn_os:command("pwd", Opts),
+
+  {1, _} = ktn_os:command("pwd; ls w4th3v3r", Opts),
+
+  Result2 = Result ++ "Hi\n",
+  {0, Result2} = ktn_os:command("pwd; echo Hi", #{}),
+
+  {0, "/\n"} = ktn_os:command("cd /; pwd").
```
Comment (elbrujohalcon) [inline test/ktn_os_SUITE.erl]: You should also test that the logging function works.

Comment (elbrujohalcon) [inline test/ktn_os_SUITE.erl]: And also that it doesn't crash if you don't provide that option in `Opts`

Comment (jfacorro) [inline test/ktn_os_SUITE.erl]: Not providing that option is tested in line 58.

Comment (elbrujohalcon) [inline test/ktn_os_SUITE.erl]: Oh, I see… you're right, sorry.

File: src/ktn_os.erl (hunk 2 above)
Comment (jfacorro) [inline src/ktn_os.erl]: I'm not sure how can we test this :confounded:

File: src/ktn_os.erl [hunk 5]
```diff
@@ -0,0 +1,49 @@
+%% @doc Utility functions to run commands in the underlying OS.
+-module(ktn_os).
+
+-export([command/1, command/2]).
+
+-type opts() :: #{log_fun => function()}.
```
Comment (elbrujohalcon) [inline src/ktn_os.erl]: can you improve the type of `log_fun`, using something like `fun((iodata()) -> _)` (or whatever applies) instead of `function()` ?

**Pull request merged**